src/Microsoft.ApiDesignGuidelines.Analyzers/Core/OperatorOverloadsHaveNamedAlternates.cs

```diff
@@ -1,69 +1,263 @@
+                case "op_Division":
+                case "op_DivisionAssignment":
+                    return createSingle("Divide");
+                case "op_Equality":
+                case "op_Inequality":
+                    return createSingle("Equals");
+                case "op_ExclusiveOr":
+                case "op_ExclusiveOrAssignment":
+                    return createSingle("Xor");
+                case "op_GreaterThan":
+                case "op_GreaterThanOrEqual":
+                case "op_LessThan":
+                case "op_LessThanOrEqual":
+                    return new ExpectedAlternateMethodGroup(alternateMethod1: "CompareTo", alternateMethod2: "Compare");
+                case "op_Increment":
+                    return createSingle("Increment");
+                case "op_LeftShift":
+                case "op_LeftShiftAssignment":
+                    return createSingle("LeftShift");
+                case "op_LogicalAnd":
+                    return createSingle("LogicalAnd");
+                case "op_LogicalOr":
+                    return createSingle("LogicalOr");
+                case "op_LogicalNot":
+                    return createSingle("LogicalNot");
+                case "op_Modulus":
+                case "op_ModulusAssignment":
+                    return new ExpectedAlternateMethodGroup(alternateMethod1: "Mod", alternateMethod2: "Remainder");
+                case "op_MultiplicationAssignment":
+                case "op_Multiply":
+                    return createSingle("Multiply");
+                case "op_OnesComplement":
+                    return createSingle("OnesComplement");
+                case "op_RightShift":
+                case "op_RightShiftAssignment":
+                case "op_SignedRightShift":
+                case "op_UnsignedRightShift":
+                case "op_UnsignedRightShiftAssignment":
+                    return createSingle("RightShift");
+                case "op_Subtraction":
+                case "op_SubtractionAssignment":
+                    return createSingle("Subtract");
+                case "op_UnaryNegation":
+                    return createSingle("Negate");
+                case "op_UnaryPlus":
+                    return createSingle("Plus");
+                case "op_Implicit":
+                case "op_Explicit":
+                    return new ExpectedAlternateMethodGroup(alternateMethod1: $"To{returnType.Name}", alternateMethod2: $"From{returnType.Name}");
+                default:
+                    return null;
+            }
+        }
+
+        internal class ExpectedAlternateMethodGroup
+        {
+            public string AlternateMethod1 { get; }
+            public string AlternateMethod2 { get; }
+
+            public ExpectedAlternateMethodGroup(string alternateMethod1, string alternateMethod2 = null)
+            {
+                AlternateMethod1 = alternateMethod1;
+                AlternateMethod2 = alternateMethod2;
```

Comment:
Huh, learn something new every day. I didn't know this would work without the property having a `private set`.